packages/scrollable_positioned_list/lib/src/scrollable_positioned_list.dart

```diff
@@ -235,6 +238,19 @@ class _ScrollablePositionedListState extends State<ScrollablePositionedList>
     super.dispose();
   }
 
+  @override
+  void didUpdateWidget(ScrollablePositionedList oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.itemCount != null) {
+      if (frontTarget > widget.itemCount - 1) {
+        frontTarget = widget.itemCount - 1;
```

Comment:
should we handle the case where itemCount is 0?

Reply:
Also, do we have a test case that covers an update where the length of the list is now shorter/

Reply:
> should we handle the case where itemCount is 0?

I think it should be handled in PositionedList instead, as `positionedIndex = -1` is semantically correct when itemCount is 0.

Reply:
> Also, do we have a test case that covers an update where the length of the list is now shorter/

I added some tests. I'm not very familiar with the widget test API, please correct it if wrong.